clients/settings/permissions/permissions_test.go

```diff
@@ -0,0 +1,791 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package permissions_test
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/clients/settings/permissions"
+)
+
+func TestNewClient(t *testing.T) {
+	actual := permissions.NewClient(&rest.Client{})
+	require.IsType(t, permissions.Client{}, *actual)
+}
+
+func TestClient_GetAllAccessors(t *testing.T) {
+	t.Run("when called without id parameter, returns an error", func(t *testing.T) {
+		client := permissions.NewClient(&rest.Client{})
+
+		actual, err := client.GetAllAccessors(t.Context(), "")
+		assert.Error(t, err)
+
+		var errPermissions permissions.ErrorPermissions
+		assert.ErrorAs(t, err, &errPermissions)
+		assert.ErrorAs(t, err, &permissions.ErrorMissingObjectID)
+		assert.Equal(t, permissions.GET, errPermissions.Operation)
+		assert.Empty(t, actual)
+	})
+
+	t.Run("successfully requesting all permissions for requested objectID", func(t *testing.T) {
+		getResponse := `{
+					"accessors": [
+						{
+						  "permissions": [
+							"r"
+						  ],
+						  "accessor": {
+							"type": "all-users"
+						  }
+						},
+						{
+						  "permissions": [
+							"r",
+							"w"
+						  ],
+						  "accessor": {
+							"type": "group",
+							"id": "4c75c5cb-4f85-4a49-811a-cdf9ae55fd4e"
+						  }
+						},
+						{
+						  "permissions": [
+							"r",
+							"w"
+						  ],
+						  "accessor": {
+							"type": "user",
+							"id": "b3d80429-98b7-44d7-b7ab-3ea453d2f18e"
+						  }
+						}
+					  ]
+					}`
+		server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			t.Log(request.URL.String())
+			require.Equal(t, http.MethodGet, request.Method)
+			require.Equal(t, "/platform/classic/environment-api/v2/settings/objects/my-object-id/permissions", request.URL.Path)
+
+			writer.WriteHeader(http.StatusOK)
+			writer.Write([]byte(getResponse))
+		}))
+		defer server.Close()
+
+		url, _ := url.Parse(server.URL)
+		client := permissions.NewClient(rest.NewClient(url, server.Client()))
+
+		resp, err := client.GetAllAccessors(t.Context(), "my-object-id")
+
+		assert.NoError(t, err)
+		assert.NotEmpty(t, resp)
+		assert.Equal(t, getResponse, string(resp.Data))
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+	})
+
+	t.Run("If settings object with ID doesn't exists on server returns error", func(t *testing.T) {
+		errorResponse := `{"error": {"code": 404,"message": "Settings not found"}}`
+
+		server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			switch request.Method {
```

Comment:
I would like to get rid of all the switch cases, as everything can be done with a server mux. This would also improve readability. No blocker for the PR though